server.py

```diff
@@ -1,43 +1,93 @@
 from io import BytesIO
 from flask import Flask, send_file, request
 from json import dumps
-from diploma import generate_diploma
-from template import valid_template_names
+from diploma import generate_diploma, UnknownFields, \
+    MissingFields, preview_template, preview_signature
+from template import import_templates, BadSignature, \
+    frontend_templates_as_json, frontend_signatures_as_json
+import os
 
 app = Flask(__name__)
 
 
-def serve_diploma_image(diploma_image):
+def get_size(width, height):
+    if not (width or height):
+        size = None
+    elif not width and height:
+        size = (height, height)
+    elif not height and width:
+        size = (width, width)
+    else:
+        size = (height, width)
+
+    if size:
+        size = tuple(int(dim) for dim in size)
+
+    return size
+
+
+def serve_image(diploma_image):
     iostream = BytesIO()
     diploma_image.save(iostream, 'PNG')
     iostream.seek(0)
     return send_file(iostream, mimetype='image/png')
 
 
-@app.route('/templates.json')
-def serve_templates():
-    return send_file('templates/templates.json')
+@app.route('/<data>.json')
+def serve_json_file(data):
+    if data == "templates":
+        return frontend_templates_as_json()
+    elif data == "signatures":
+        return frontend_signatures_as_json()
+
+
+@app.route('/preview/<filename>')
```

Comment:
Isn't it better if we have the preview paths for images and signatures separated?
E.g. Preview a template: `/preview/template/<name>` and preview a signature: `/preview/signature/<name>`.
This will make it more future-proof and we can more easily add support for previewing other things as well, without breaking changes to the API.
It also reduces the chance of a name conflict.